plotly/tools.py

```diff
@@ -3305,7 +3305,7 @@ def _trisurf(x, y, z, simplices, show_colorbar, colormap=None,
                         color_func[index] = FigureFactory._label_rgb(foo)
 
             mean_dists = np.asarray(color_func)
-        else:
+        elif hasattr(color_func, '__call__'):
```

Comment:
> Why not just elif color_func is not None?

Your way makes sense upon closer inspection. Actually, I think `else` would work as well, since if `color_func` was a function, it would therefore not be `None` nor a `list/array`